dnf5daemon-server/services/rpm/rpm.cpp

```diff
@@ -392,6 +408,64 @@ sdbus::MethodReply Rpm::list(sdbus::MethodCall & call) {
     return reply;
 }
 
+void Rpm::list_fd(sdbus::MethodCall & call) {
+    // read options from dbus call
+    dnfdaemon::KeyValueMap options;
+    call >> options;
+    // get the output pipe file descriptor
+    sdbus::UnixFd dbus_unix_fd;
+    call >> dbus_unix_fd;
+    // id of the output filedescriptor
+    std::string out_fd_id;
+    call >> out_fd_id;
+
+    int out_fd = dbus_unix_fd.get();
+
+    session.fill_sack();
+
+    auto query = filter_packages(options);
+
+    std::vector<std::string> package_attrs =
+        dnfdaemon::key_value_map_get<std::vector<std::string>>(options, "package_attrs", std::vector<std::string>());
+    std::string error_msg{};
+    for (const auto & pkg : query) {
+        std::string jsonstr;
+        try {
+            jsonstr = package_to_json(pkg, package_attrs) + "\n";
+        } catch (const std::exception & ex) {
+            error_msg = fmt::format(
+                "Error serializing package \"{0}\" from repo \"{1}\": {2}",
+                pkg.get_nevra(),
+                pkg.get_repo_id(),
+                ex.what());
+        } catch (...) {
```

Comment:
Interesting - Do you have any particular experience that this might happen?

Reply:
Nothing in particular. Just didn't want the server to abort.
But the unknown exceptions are already caught one level higher in `ThreadsManager::handle_method_fd()` so I will remove these "catch all" branches here.